### DISCUSSION TITLE
`wait` strategy used with custom `Snapshotting` extension?

### DISCUSSION THREAD

**Discussion by simondelphia** [pointfreeco/swift-snapshot-testing#543] [Q&A]
If you have a `Snapshotting` extension that takes in whatever type and transforms it to a `UIView` and try to use `wait` with it to try to capture some delayed change to the view, it doesn't seem to have the desired effect.

Here I used an `Int` and some view `MyView` which has an initializer that accepts an `Int` as input just as an example. Let's say `MyView` animates some label such that it animates from a default state to one showing the passed value.

```
extension Snapshotting where Value == Int, Format == UIImage {
    static let image: Self = Snapshotting<UIView, UIImage>.image.pullback { 
        MyView(value: $0) // + some appropriate constraints to give it some size
    }
}
```

```
func testMyView() {
    assertSnapshot(
        matching: 10,
        as: .wait(for: 2, on: .image)
    )
}
```

The view is produced but the resulting snapshot is as if there was no wait. If Instead I don't use a `Snapshotting` extension and just call `assertSnapshot` on a `MyView` instance directly, it seems to work properly. 

Plus if you put a long delay on the `wait` and put a breakpoint inside the `pullback`, the breakpoint is only hit after the delay, so the view isn't getting initialized until after the delay and thus the `wait` is pointless here.

Is this expected behaviour? Are you supposed to do something differently for `wait` when using `pullback` in a `Snapshotting` extension?

**Comment by doozMen** [comment]
still have this issue
